### PR TITLE
Android: Simplify Custom registration

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

**Review by Archifer — APPROVED** [review]

**Review by aleknow — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/RegistrationManager.kt [hunk 1]
```diff
@@ -2,19 +2,15 @@ package com.onegini.mobile.sdk.reactnative.managers
...
-import com.onegini.mobile.sdk.reactnative.handlers.registration.RegistrationRequestHandler
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class RegistrationManager @Inject constructor(
-    private val oneginiSDK: OneginiSDK,
-    private val registrationRequestHandler: RegistrationRequestHandler
-) {
+class RegistrationManager @Inject constructor(private val oneginiSDK: OneginiSDK) {
```
[the remark this answers is not in the document]
Comment (MichielVrins) [reply android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/RegistrationManager.kt]: That seems entirely reasonsable and I have updated that.

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/exception/OneginiReactNativeException.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
 package com.onegini.mobile.sdk.reactnative.exception
 
-class OneginiReactNativeException(var errorType: Int, message: String): Exception(message)
\ No newline at end of file
+class OneginiReactNativeException constructor(var errorType: Int, message: String): Exception(message) {
+    constructor(error: OneginiWrapperErrors) : this(error.code, error.message)
+}
```
[the remark this answers is not in the document]
Comment (MichielVrins) [reply android/src/main/java/com/onegini/mobile/sdk/reactnative/exception/OneginiReactNativeException.kt:5]: https://onewelcome.atlassian.net/browse/RNP-155

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/RegistrationManager.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichielVrins) [reply android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/RegistrationManager.kt]: I did name it CustomRegistrationActionManager as this seems more descriptive

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/CustomRegistrationActionManager.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        return null
+    }
+
+    fun getCustomRegistrationActions(): ArrayList<CustomRegistrationAction> {
+        return customRegistrationActions
+    }
+
+    fun addCustomRegistrationAction(action: CustomRegistrationAction) {
+        customRegistrationActions.add((action))
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/CustomRegistrationActionManager.kt:26]: duplicated `()`

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/CustomRegistrationActionManager.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    private val customRegistrationActions = ArrayList<CustomRegistrationAction>()
+
+    fun getCustomRegistrationAction(id: String?): CustomRegistrationAction? {
+        for (action in customRegistrationActions) {
+            if (action.id == id) {
+                return action
+            }
+        }
+        return null
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/CustomRegistrationActionManager.kt:18]: you could simplify it to something like this:
`return customRegistrationActions.firstOrNull { it.id == id }`

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/CustomRegistrationActionManager.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        customRegistrationActions.add((action))
+    }
+
+    fun getActiveCustomRegistrationAction(): CustomRegistrationAction? {
+        customRegistrationActions.forEach { action ->
+            if (action.isInProgress()){
+                return action
+            }
+        }
+        return null
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/managers/CustomRegistrationActionManager.kt:35]: this one could also be simplified:
`customRegistrationActions.firstOrNull { it.isInProgress() }`

**Review by aleknow — CHANGES_REQUESTED** [review]

**Review by aleknow — APPROVED** [review]

**Pull request merged**